client/src/test/java/io/pravega/client/netty/impl/ConnectionFactoryImplTest.java

```diff
@@ -106,8 +107,12 @@ public void initChannel(SocketChannel ch) throws Exception {
 
     @After
     public void tearDown() throws Exception {
-        serverChannel.close();
-        serverChannel.closeFuture();
+        serverChannel.close().awaitUninterruptibly();
+        bossGroup.shutdownGracefully().await();
+        workerGroup.shutdownGracefully().await();
+        if (sslCtx != null) {
+            ReferenceCountUtil.safeRelease(sslCtx);
```

Comment:
Do we really need to release the SSL Context explicitly here? Could you please describe the reason? 

For what it's worth, the TLS example from Netty also doesn't do it: 
https://github.com/netty/netty/blob/4.1/example/src/main/java/io/netty/example/securechat/SecureChatServer.java

Reply:
That **sslCtx** is an [ReferenceCountedOpenSslContext](https://netty.io/4.1/api/io/netty/handler/ssl/ReferenceCountedOpenSslContext.html)
and it needs an explicit release.
The example you are linking is very trivial and maybe it is not using OpenSSL